php/elFinderVolumeOneDrive.class.php

```diff
@@ -0,0 +1,1797 @@
+        
+		return true;
+	}
+    
+    /**
+     * Get script url
+     * 
+     * @return string full URL
+     * @author Naoki Sawada
+     */
+    private function getConnectorUrl()
+    {
+        $url  = ((isset($_SERVER['HTTPS']) && strtolower($_SERVER['HTTPS']) !== 'off')? 'https://' : 'http://')
+               . $_SERVER['SERVER_NAME']                                              // host
+              . ($_SERVER['SERVER_PORT'] == 80 ? '' : ':' . $_SERVER['SERVER_PORT'])  // port
+               . $_SERVER['REQUEST_URI'];                                             // path & query
+        list($url) = explode('?', $url);
+        
+        return $url;
+    }
+    
+    /*********************************************************************/
+    /*                        INIT AND CONFIGURE                         */
+    /*********************************************************************/
+
+    /**
+     * Prepare FTP connection
+     * Connect to remote server and check if credentials are correct, if so, store the connection id in $ftp_conn
+     *
+     * @return bool
+     * @author Dmitry (dio) Levashov
+     * @author Cem (DiscoFever)
+     **/
+    protected function init()
+    {
+		if (!$this->options['accessToken']) {		
+            return $this->setError('Required options undefined.');
+        }
+        
+        // make net mount key
+        $this->netMountKey = md5(join('-', array('onedrive', $this->options['path'])));
+        
+        if (! $this->onedrive) {					
+            try {
+                $this->onedrive = json_decode($this->options['accessToken']);
+                if (true !== ($res = $this->refreshOneDriveToken($this->onedrive))) {
+                    return $this->setError($res);
+                }
+            } catch (InvalidArgumentException $e) {
+                return $this->setError($e->getMessage());
+            }
+            try {
+				$this->onedrive = json_decode($this->options['accessToken']);
+            } catch (Exception $e) {
+                return $this->setError($e->getMessage());
+            }
+        }
+        
+        if (! $this->onedrive) {		
+            return $this->setError('OAuth extension not loaded.');
+        }
+        
+        // normalize root path
+        if ($this->options['path'] == 'root') {
+            $this->options['path'] = '/';
+        }
+		
+        $this->root = $this->options['path'] = $this->_normpath($this->options['path']);
+        
+        $this->options['root'] == '' ?  $this->options['root']= 'OneDrive.com' : $this->options['root'];
+        
+        if (empty($this->options['alias'])) {
+		    $this->options['alias'] = ($this->options['path'] === '/') || ($this->options['path'] === 'root')? $this->options['root'] : $this->getPathtoName($this->options['path']).'@OneDrive';
+        }
+
+        $this->rootName = $this->options['alias'];
+               
+        $this->tmbPrefix = 'onedrive'.base_convert($this->root, 10, 32);
```

Comment:
after checked your demo page mount with google drive,  tmb  create only one thumbnail for same name volume